File: algoexpert/SortAlgorithms/bubbleSort.js
// Write a function that takes in an array of integers and
// returns a sorted version of that array. Use the Bubble
// Sort algorithm to sort the array.

function bubbleSort(array) {
    let isSorted = false;
    let counter = 0;

    while (!isSorted) {
        isSorted = true;
        for (let i = 0; i < array.length - 1 - counter; i++) {
            if (array[i] > array[i + 1]) {
                swap(i, i + 1, array);
                isSorted = false;
            }
        }
        counter++;
    }
    return array;
}

function swap(i, j, array) {
    let temp = array[j];
    array[j] = array[i];
    array[i] = temp;
}

File: AlgoExpert/Python/Sorting/BubbleSort.py
def bubbleSort(array):
    isSorted = False
    counter = 0

    while not isSorted:
        for idx in range(len(array) - 1 - counter):
            if array[idx] > array[idx + 1]:
                swap(idx, idx + 1, array)
                isSorted = False
        counter += 1
    return array


def swap(i, j, array):
    array[i], array[j] = array[j], array[i]
